src/MercadoPago/Core/Plugin/OrderCancelPlugin.php

```diff
@@ -65,6 +65,11 @@ protected function salesOrderBeforeCancel()
         }
 
         $orderStatus = $this->order->getData('status');
+
+        if(!isset($this->order->getPayment()->getData('additional_information')['status'])){
+            return ;
+        }
+
         $orderPaymentStatus = $this->order->getPayment()->getData('additional_information')['status'];
 
         $paymentID = $this->order->getPayment()->getData('additional_information')['payment_id_detail'];
```

Comment:
In my case was necessary check this index.
       if(!isset($this->order->getPayment()->getData('additional_information')['payment_id_detail'])){
            return ;
        }